src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -399,18 +405,48 @@ private function buildConstructor(ClassBuilder $class): void
 ', ['PROPERTY' => $p->getName(), 'ORG_NAME' => $p->getOriginalName()]);
         }
 
-        $body .= '
+        if ($class->shouldAllowExtraKeys()) {
+            $body .= '
+    $this->_extraKeys = $value;
+';
+        } else {
+            $body .= '
     if ([] !== $value) {
         throw new InvalidConfigurationException(sprintf(\'The following keys are not supported by "%s": \', __CLASS__).implode(\', \', array_keys($value)));
     }';
 
-        $class->addUse(InvalidConfigurationException::class);
+            $class->addUse(InvalidConfigurationException::class);
+        }
+
         $class->addMethod('__construct', '
 public function __construct(array $value = [])
 {
 '.$body.'
-}
-');
+}');
+    }
+
+    private function buildSetExtraKey(ClassBuilder $class): void
+    {
+        if (!$class->shouldAllowExtraKeys()) {
+            return;
+        }
+
+        $class->addProperty('_extraKeys');
+
+        $class->addMethod('set'.substr($class->getName(), 0, -6).'ExtraKey', '
+/**
+ * @return $this
+ */
+public function NAME(string $key, $value): self
+{
+    if (null === $value) {
+        unset($this->_extraKeys[$key]);
```

Comment:
Do we need this? Maybe `null` is a valid value?

Reply:
@Nyholm I was going for consistency with the "non-extra" keys. Currently every property that's `null` doesn't get returned, eg:

```php
if (null !== $this->baz) {
    $output['baz'] = $this->baz;
}
```

Reply:
You are correct. Let's keep it as it is.